keras_contrib/layers/convolutional.py

```diff
@@ -456,6 +458,101 @@ def get_config(self):
 get_custom_objects().update({"CosineConvolution2D": CosineConvolution2D})
 get_custom_objects().update({"CosineConv2D": CosineConv2D})
+
+
+class SubPixelUpscaling(Layer):
+    """ Sub-pixel convolutional upscaling layer based on the paper "Real-Time Single Image
+    and Video Super-Resolution Using an Efficient Sub-Pixel Convolutional Neural Network"
+    (https://arxiv.org/abs/1609.05158).
+
+    This layer requires a Convolution2D prior to it, having output nb_filter computed according to
+    the formula :
+
+        nb_filter = k * (scale_factor * scale_factor)
+        where k = a user defined number of filters (generally larger than 32)
+              scale_factor = the upscaling factor (generally 2)
+
+    This layer performs the depth to space operation on the convolution filters, and returns a
+    tensor with the size as defined below.
+
+    # Note:
+        This layer does not work with mismatched backend and dim ordering.
+        For example, TH dim ordering with Tensorflow backend / TF dim ordering with Theano backend.
+
+    # Example :
+    ```python
+        # A standard subpixel upscaling block
+        x = Convolution2D(256, 3, 3, border_mode='same', activation='relu')(...)
+        u = SubPixelUpscaling(scale_factor=2)(x)
+
+        [Optional]
+        x = Convolution2D(256, 3, 3, border_mode='same', activation='relu')(u)
+    ```
+
+        In practice, it is useful to have a second convolution layer after the
+        SubPixelUpscaling layer to speed up the learning process.
+
+        However, if you are stacking multiple SubPixelUpscaling blocks, it may increase
+        the number of parameters greatly, so the Convolution layer after SubPixelUpscaling
+        layer can be removed.
+
+    # Arguments
+        scale_factor: Upscaling factor.
+        dim_ordering: Can be 'th' or 'tf'. Note: mismatched dim ordering will
+                      cause a ValueError to be raised.
+
+    # Input shape
+        4D tensor with shape:
+        `(samples, k * (scale_factor * scale_factor) channels, rows, cols)` if dim_ordering='th'
+        or 4D tensor with shape:
+        `(samples, rows, cols, k * (scale_factor * scale_factor) channels)` if dim_ordering='tf'.
+
+    # Output shape
+        4D tensor with shape:
+        `(samples, k channels, rows * scale_factor, cols * scale_factor))` if dim_ordering='th'
+        or 4D tensor with shape:
+        `(samples, rows * scale_factor, cols * scale_factor, k channels)` if dim_ordering='tf'.
+
+    """
+
+    def __init__(self, scale_factor=2, dim_ordering='default', **kwargs):
+        super(SubPixelUpscaling, self).__init__(**kwargs)
+
+        self.scale_factor = scale_factor
+        self.dim_ordering = dim_ordering
+
+        if self.dim_ordering == 'default':
+            self.dim_ordering = K.image_dim_ordering()
+
+        if (K.backend() == 'theano' and self.dim_ordering == 'tf') or \
+                (K.backend() == 'tensorflow' and self.dim_ordering == 'th'):
```

Comment:
Hmm let me try it. When I did it manually, it crashed with tensor errors which I couldn't debug properly.